flytepropeller/pkg/controller/nodes/task/k8s/plugin_collector.go

```diff
@@ -45,36 +45,30 @@ type ResourceLevelMonitor struct {
 	once sync.Once
 }
 
-// The reason that we use namespace as the one and only thing to cut by is because it's the feature that we are sure that any
-// K8s resource created by a plugin will have (as yet, Flyte doesn't have any plugins that create cluster level resources and
-// it probably won't for a long time). We can't assume that all the operators and CRDs that Flyte will ever work with will have
-// the exact same set of labels or annotations or owner references. The only thing we can really count on is namespace.
-func (r *ResourceLevelMonitor) countList(ctx context.Context, objects []interface{}) map[string]int {
-	// Map of namespace to counts
-	counts := map[string]int{}
-
-	// Collect the object counts by namespace
-	for _, v := range objects {
-		metadata, err := meta.Accessor(v)
-		if err != nil {
-			logger.Errorf(ctx, "Error converting obj %v to an Accessor %s\n", v, err)
-			continue
-		}
-		counts[metadata.GetNamespace()]++
-	}
-
-	return counts
-}
-
 // The context here is expected to already have a value for the KindKey
 func (r *ResourceLevelMonitor) collect(ctx context.Context) {
 	// Emit gauges at the namespace layer - since these are just K8s resources, we cannot be guaranteed to have the necessary
 	// information to derive project/domain
-	objects := r.sharedInformer.GetStore().List()
-	counts := r.countList(ctx, objects)
+	list := metav1.PartialObjectMetadataList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       r.gvk.Kind,
+			APIVersion: r.gvk.Version,
+		},
+	}
+	if err := r.cache.List(ctx, &list); err != nil {
+		logger.Warnf(ctx, "failed to list objects for %s.%s: %v", r.gvk.Kind, r.gvk.Version, err)
```

Comment:
@hamersaw I'm seeing this log line a lot since we upgraded to 1.12 (45k times in 5 days). For instance:

```console
failed to list objects for PyTorchJob.v1: no matches for kind "PyTorchJob" in version "v1"
```

The CRD does exist in the cluster:

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  ...
  name: pytorchjobs.kubeflow.org
spec:
  group: kubeflow.org
  names:
    kind: PyTorchJob
    listKind: PyTorchJobList
    plural: pytorchjobs
    singular: pytorchjob
  scope: Namespaced
  versions:
  - ...
    name: v1
```

Tasks using the pytorch plugin also work.

Do you know what the reason could be for the cache not being aware of the pytorchjob crd?